### DISCUSSION TITLE
#New Features: Bridging nanomsg with nanomq

### DISCUSSION THREAD

**Discussion by JaylinYu** [nanomq/nanomq#298] [Ideas]
Since we have managed to make nanomq compatible with nanomsg/nng, the new bridging feature comes to my mind:
It would be great to let classical nanomsg/nng client talks with nanomq, and convert the msg into MQTT format to achieve edge-cloud collaboration. MQTT is a great protocol, however it has limited messaging patterns which nanomsg/nng could really shines.

Definetly, there must be a rule of this feature to define how to convert nanomsg <-> mqtt. It is not decided yet.

=============================

NanoMQ与NNG项目及其维护者一直保持着良好合作，希望能够将brokerless和broker的两种消息模式在边缘统一，提供一个高性能的消息总线。但nanomsg和nanomq的桥接需要社区群策群力提供各种意见。希望各位mqtt和nanomsg的使用者能够提供一些应用场景和桥接需求供我参考

**Comment by JaylinYu** [comment]
hi all, the bridging module of nng/nanomq is already done. Now nanomq supports two way channl that can convert msg between nanomsg/nng and mqtt

**Reply by mik1234mc** [reply]
Hi Jaylin, is there any reference code example how to use nanomsg/nng bridging with nanomq(mqtt)?

**Reply by JaylinYu** [reply]
Please check nngproxy. I will add tutorials with example code.
